### PR TITLE
visual-studio 17.5.6.3

### PR DESCRIPTION
**Pull request by i0ntempest**
Created with `brew bump-cask-pr`.

### CODE REVIEW

**Comment by i0ntempest** [comment]
Need some help updating the livecheck, and possibly other stuff.

**Comment by i0ntempest** [comment]
The installer is older than the publicly available version available as an app in a standalone dmg @bevanjkay

**Comment by bevanjkay** [comment]
@i0ntempest Is there a link to the standalone dmg from upstream's website, their download links point to the installer.

**Comment by i0ntempest** [comment]
> @i0ntempest Is there a link to the standalone dmg from upstream's website, their download links point to the installer.

I have changed the link in the file, it now points to the standalone dmg.

**Comment by bevanjkay** [comment]
How did you find the link to the standalone dmg? Because we need to find a way to reference the version for `livecheck`.

**Comment by i0ntempest** [comment]
> How did you find the link to the standalone dmg? Because we need to find a way to reference the version for `livecheck`.

It’s from MacUpdater. It may also have a working livecheck link, I’ll see when I get back to home.

**Comment by razvanazamfirei** [comment]
This is the arm link from macupdater: 
`https://download.visualstudio.microsoft.com/download/pr/91b81b8c-74ac-4710-a93f-49bf4553845e/ed3a76bd92fcbfe800b830b2739774e0/visualstudioformac-17.5.6.3-arm64.dmg`

Their history link is: `https://learn.microsoft.com/en-us/visualstudio/releases/2022/mac-release-notes` which did not seem helpful.

**Comment by i0ntempest** [comment]
Any way we can move forward on this?

**Comment by bevanjkay** [comment]
We need a way for livecheck to check that the latest version is correct.
From upstream's website I can find a download link for the installer.

**Comment by razvanazamfirei** [comment]
I found a better livecheck source and I'm rewriting this now.

File: Casks/visual-studio.rb [hunk 1]
```diff
@@ -1,20 +1,27 @@
...
 
   livecheck do
-    url "https://aka.ms/vs/mac/download"
-    regex(%r{/download/pr/([^/]+)/([^/]+)/visualstudioformacinstaller[._-]v?(\d+(?:\.\d+)+).dmg}i)
-    strategy :header_match do |headers, regex|
-      match = headers["location"].match(regex)
-      next if match.blank?
-
-      "#{match[3]},#{match[1]},#{match[2]}"
+    url "https://aka.ms/vsmac/manifest/17-stable"
```
Comment (bevanjkay) [inline Casks/visual-studio.rb:19]: @razvanazamfirei Should we use the below, would require a manual bump for major version

```suggestion
    url "https://aka.ms/vsmac/manifest/#{version.major}-stable"
```

Comment (razvanazamfirei) [reply Casks/visual-studio.rb]: @bevanjkay, sure! I hope it's still going to work with a new version, but worth a shot!

**Review by razvanazamfirei — APPROVED** [review]

**Pull request merged**